src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import java.util.Scanner;
+/**
+ * Represents a parser of user commands in Duke chat bot.
+ */
+public class Parser {
+    protected Scanner sc;
+
+    public Parser(Scanner sc) {
+        this.sc = sc;
+    }
+
+    /**
+     * Reads and react to user command
+     * @param tasks Current task list.
+     */
+    public void read(TaskList tasks) {
+        while(true) {
+            String cmd = sc.nextLine();
+            Scanner cmdSc = new Scanner(cmd);
+            if (cmd.toLowerCase().equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
```

Comment:
Why not refactor this to be done by the UI instead, so all printing is done by the UI?